test/test_convert_cache.py

```diff
@@ -0,0 +1,102 @@
+import json
+from pathlib import Path
+from shutil import copyfile
+
+import pytest
+import jsonschema
+import kernel_tuner
+
+from kernel_tuner.cache.convert import convert_cache_file
+from kernel_tuner.cache.convert import VERSIONS
+
+KERNEL_TUNER_PATH = Path(kernel_tuner.__file__).parent
+TEST_PATH         = Path(__file__).parent
+
+# Mock schema files
+MOCK_CONVERT_PATH = TEST_PATH / "test_convert_files"
+MOCK_CONVERT_FILE = MOCK_CONVERT_PATH / "mock_cache.json"
+
+MOCK_SCHEMAS_PATH = MOCK_CONVERT_PATH / "mock_schemas"
+MOCK_SCHEMA_OLD   = MOCK_SCHEMAS_PATH / "1.0.0/schema.json"
+MOCK_SCHEMA_NEW   = MOCK_SCHEMAS_PATH / "1.2.0/schema.json"
+
+# Actual schema files
+CONVERT_PATH      = TEST_PATH / "test_cache_files"
+CONVERT_FILE      = CONVERT_PATH / "small_cache.json"
+
+SCHEMAS_PATH      = KERNEL_TUNER_PATH / "schema/cache"
+SCHEMA_OLD        = SCHEMAS_PATH / VERSIONS[ 0] / "schema.json"
+SCHEMA_NEW        = SCHEMAS_PATH / VERSIONS[-1] / "schema.json"
+
+
+
+class TestConvertCache:
+    # Test conversion system using mock schema and cache files
+    def test_convert_mock(self, tmp_path):
```

Comment:
Maak de naam van de test method langer zodat er geen comment nodig is. Zelfdocumenterende code is bliss 😁